storage/remote/generic/generic.go

```diff
@@ -62,16 +69,15 @@ func (c *Client) Store(samples model.Samples) error {
 		}
 		req.Timeseries = append(req.Timeseries, ts)
 	}
-	client := NewGenericWriteClient(c.conn)
 	ctxt, _ := context.WithTimeout(context.Background(), c.timeout)
```

Comment:
Ok, seems like I can use https://godoc.org/google.golang.org/grpc#DialContext for that, although it's labeled as anexperimental API. The docstring of that function is still a bit confusing to me: "Once this function returns, the cancellation and expiration of ctx will be noop.". But the context is also attached to the returned `clientConn` object, so it does have relevance even after `DialContext` returns.

Just so I understand you correctly: So I pass a base context into there, then create a cancellable context based on that for each `Write()`? What do you mean with calling cancel "after we are done"? After the `Write()` completes (why, since it's already done then?)?